File: cesta/IvanRC-Cesta/IvanRC-Cesta-Java/src/main/java/FactoriesCesta/FactoryProducto.java
package FactoriesCesta;

import ModelsCesta.Producto;

public class FactoryProducto {

    public FactoryProducto(){}
    private static FactoryProducto instance = null;

    public static FactoryProducto getInstance(){
        if(instance == null){
            instance = new FactoryProducto();
        }
        return instance;
    }
    private static int contadorProducto = 1;
    private int nextId(){
        return contadorProducto++;
    }

    /**
     * función que sirve para crear un producto aleatoriamente
     * @return el producto cuyas propiedades se crearon al azar
     */
    public Producto crearProductoRandom() {
        final double[] precioUnitario = {12.0,2.32,4.3,4.56,76.44,22.0,12.43,24.3,8.56,7.44,16.0, 34.32,4.44,7.54,7.84,9.74,8.45,3.24,15.98,6.09,4.99};
        final String[] nombreProductos = {"gasolina", "coche", "moto", "camión", "mira", "telefono", "gato", "deberes", "sofa", "silla", "chalet", "garrafa", "garra", "portatil", "procesador", "rueda", "carrusel", "pinguino", "pera", "manzana", "aguacate"};
        return new Producto(nextId(), precioUnitario[(int) (Math.random()*precioUnitario.length)], nombreProductos[(int) (Math.random()*nombreProductos.length)]);
    }
}

File: cesta/IvanRC-Cesta/IvanRC-Cesta-Kotlin/src/main/kotlin/FactoriesCestaDeCompra/FactoryProducto.kt
package FactoriesCestaDeCompra

import ModelsCestaDeCompra.Producto

class FactoryProducto {

    companion object{
        private var instance: FactoryProducto? = null

        fun getInstance(): FactoryProducto?{
            if(instance == null){
                instance = FactoryProducto()
            }
            return instance
        }
        var contadorProductos = 1
        fun nextId():Int{
            return contadorProductos++
        }
    }

    /**
     * función que sirve para crear un producto aleatoriamente
     * @return el producto cuyas propiedades se crearon al azar
     */
    fun crearProductoRandom(): Producto {
        val precioUnitario = arrayOf(12.0,2.32,4.3,4.56,76.44,22.0,12.43,24.3,8.56,7.44,16.0, 34.32,4.44,7.54,7.84,9.74,8.45,3.24,15.98,6.09,4.99)
        val nombreProductos = arrayOf("gasolina", "coche", "moto", "camión", "mira", "telefono", "gato", "deberes", "sofa", "silla", "chalet", "garrafa", "garra", "portatil", "procesador", "rueda", "carrusel", "pinguino", "pera", "manzana", "aguacate")
        return Producto(nextId(), precioUnitario.random(), nombreProductos.random())
    }
}